packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -31,11 +32,13 @@ import 'message_codecs.dart';
 class BasicMessageChannel<T> {
   /// Creates a [BasicMessageChannel] with the specified [name], [codec] and [binaryMessenger].
   ///
-  /// None of [name], [codec], or [binaryMessenger] may be null.
-  const BasicMessageChannel(this.name, this.codec, { this.binaryMessenger = defaultBinaryMessenger })
-    : assert(name != null),
-      assert(codec != null),
-      assert(binaryMessenger != null);
+  /// [name], [codec] cannot be null. A default [BinaryMessenger] instance will
+  /// be used if [binaryMessenger] is null.
+  BasicMessageChannel(this.name, this.codec, { this.binaryMessenger }) {
+    assert(name != null);
+    assert(codec != null);
+    binaryMessenger ??= ServicesBinding.instance.provideBinaryMessenger();
+  }
```

Comment:
we need to keep this class `const`.

Easiest way to do this, I think, is to have the default value of the `binaryMessenger` argument be null, have it set a private field `_binaryMessenger`, and then have `BinaryMessenger get binaryMessenger => _binaryMessenger ?? ServicesBinding.instance.defaultBinaryMessenger`, where the latter is a getter that returns a precached value created during `initInstances`.

Reply:
Thanks! I'll try this approach.

Reply:
Updated accordingly. PTAL.